src/token.rs

```diff
@@ -290,6 +331,37 @@ fn split_label_text(input: &str) -> Option<(&str, &str)> {
     }
 }
 
+/// assert_eq!(sgf_parser::parse_outcome_str("B+R"), Outcome::WinnerByResign(Color::Black));
+/// assert_eq!(sgf_parser::parse_outcome_str("B+35.0"), Outcome::WinnerByPoints(Color::Black, 35.0));
+/// assert_eq!(sgf_parser::parse_outcome_str("W+R"), Outcome::WinnerByResign(Color::White));
+/// assert_eq!(sgf_parser::parse_outcome_str("W+66.5"), Outcome::WinnerByPoints(Color::Black, 65
+/// .5));
+///
+fn parse_outcome_str(s: &str) -> Result<Outcome, SgfError> {
+    if s.is_empty() {
+        return Err(SgfError::from(SgfErrorKind::ParseError));
+    }
+    let winner_option: Vec<&str> = s.split("+").collect();
+    if winner_option.len() != 2 {
+        return Err(SgfError::from(SgfErrorKind::ParseError));
+    }
+    assert_eq!(winner_option.len(), 2);
+    let winner = if winner_option[0].starts_with("B") {
+        Black
+    } else {
+        White
+    };
+
+    let outcome = if winner_option[1].starts_with("R") {
+        WinnerByResign(winner)
+    } else {
+        WinnerByPoints(winner,
+                       winner_option[1].parse::<f32>().expect("Parse the result score"),
```

Comment:
When we encounter an invalid number, we should return a `SgfErrorKind::ParseError.